src/selectedtests/app/test_mapping_work_item.py

```diff
@@ -0,0 +1,120 @@
+        created_on: datetime,
+        project: str,
+        source_file_regex: str,
+        test_file_regex: str,
+        module: str,
+        module_source_file_regex: str,
+        module_test_file_regex: str,
+    ):
+        """
+        Create a test_mapping work item.
+
+        :param start_time: Time work was started on item.
+        :param end_time: Time work completed on item.
+        :param created_on: Time work item was created.
+        :param project: The name of the evergreen project to analyze.
+        :param source_file_regex: Regex pattern to match changed source files against.
+        :param test_file_regex: Regex pattern to match changed test files against.
+        :param module: The name of the module to analyze.
+        :param module_source_file_regex: Regex pattern to match changed module source files against.
+        :param module_test_file_regex: Regex pattern to match changed module test files against.
+        """
+        self.start_time = start_time
+        self.end_time = end_time
+        self.created_on = created_on
+        self.project = project
+        self.source_file_regex = source_file_regex
+        self.test_file_regex = test_file_regex
+        self.module = module
+        self.module_source_file_regex = module_source_file_regex
+        self.module_test_file_regex = module_test_file_regex
+
+    @classmethod
+    def new_test_mappings(
+        cls,
+        project: str,
+        source_file_regex: str,
+        test_file_regex: str,
+        module: str = "",
+        module_source_file_regex: str = "",
+        module_test_file_regex: str = "",
+    ):
+        """
+        Create a new work item.
+
+        :param project: The name of the evergreen project to analyze.
+        :param source_file_regex: Regex pattern to match changed source files against.
+        :param test_file_regex: Regex pattern to match changed test files against.
+        :param module: The name of the module to analyze.
+        :param module_source_file_regex: Regex pattern to match changed module source files against.
+        :param module_test_file_regex: Regex pattern to match changed module test files against.
+        :return: TestMappingWorkItem instance for work item.
+        """
+        return cls(
+            None,
+            None,
+            datetime.utcnow(),
+            project,
+            source_file_regex,
+            test_file_regex,
+            module,
+            module_source_file_regex,
+            module_test_file_regex,
+        )
+
+    def insert(self, collection):
+        """
+        Add this work item to the Mongo collection.
+
+        If this work item already exists, just update the created_on timestamp.
+
+        :param collection: Mongo collection containing queue.
+        :return: True if item was new record was added to collection.
+        """
+        LOGGER.info(
+            f"Adding new test_mapping work item for {self.project}, {self.source_file_regex}",
```

Comment:
[nit] I tend avoid f-strings with structlog. It is a personal preference, but I find if there is something I want to add as a f-string variable, it might be more useful to add it as a named-parameter. Because of how named parameters show up in the log, it gives a nice anchor to search on if needed.